src/matrices.jl

```diff
@@ -169,7 +171,14 @@ Quantification Analysis. Theory and Best Practices*, Springer, pp. 3-43 (2015).
 """
 function RecurrenceMatrix(x, ε; metric = DEFAULT_METRIC, kwargs...)
     m = getmetric(metric)
-    s = resolve_scale(x, m, ε; kwargs...)
+    if !haskey(kwargs, :scale)
+        s = resolve_scale(x, x, m, ε; kwargs...)
+    elseif kwargs[:scale] == "powerlaw"
```

Comment:
Should this be replaced with a type-based dispatch system?  I find that it's immensely helpful to specify options, and clarify dispatches...

Reply:
I am not sure how helpful this will be in the end of the road, because it would make the code much larger. How do you think it would be done to keep the code the same size?

In addition, from the user perspective `method = "linear"` or `method = "cubic"` is simpler to understand than `method = Linear()`, `method = Cubic()`. Everyone knows strings, but not everyone knows singleton types and their role in multiple dispatch.

But in general I also prefer using multiple dispatch instead of unending `if` statements.